library/forms.inc

```diff
@@ -52,7 +53,12 @@ function getFormByEncounter(
         $sql .= " and therapy_group_id = ? and pid IS NULL ";
     }
 
-    array_push($arraySqlBind, $attendant_id);
+    if ($form_id != '') {
+        $sql .= " and id=?";
+        array_push($arraySqlBind, $attendant_id, $form_id);
+    } else {
+        array_push($arraySqlBind, $attendant_id);
+    }
```

Comment:
should not need the changes in this script (see above posts for details)

Reply:
reverted it ..